### PR TITLE
Create flyover rs builder

### PR DESCRIPTION
**Pull request by julia-zack**
-Add FlyoverRedeemScriptBuilder class (interface and implementation) to add a flyover prefix to an existing redeem script, following what is done in [FastBridgeRedeemScriptParser](https://github.com/rsksmart/bitcoinj-thin/blob/master/src/main/java/co/rsk/bitcoinj/script/FastBridgeRedeemScriptParser.java#L39) class

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptCreationException.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package co.rsk.peg.bitcoin;
+
+public class FlyoverRedeemScriptCreationException extends RuntimeException {
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptCreationException.java:3]: Maybe simply reuse RedeemScriptCreationException? I think it covers what we need

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImplTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            () -> flyoverRedeemScriptBuilder.addFlyoverDerivationHashToRedeemScript(flyoverDerivationHash, redeemScript));
+
+        assertEquals(INVALID_FLYOVER_DERIVATION_HASH, exception.getReason());
+
+        String expectedMessage = "Provided flyover derivation hash is invalid.";
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    private static Stream<Arguments> invalidDerivationHashArgsProvider() {
+        return Stream.of(Arguments.of(null, zeroHash));
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImplTest.java:47]:
```suggestion
    private static Stream<Keccak256> invalidDerivationHashArgsProvider() {
        return Stream.of(null, zeroHash);
```
No need to use Arguments if we already know the type

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImplTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        // act
+        Script redeemScriptWithFlyoverDerivationHash = flyoverRedeemScriptBuilder.addFlyoverDerivationHashToRedeemScript(flyoverDerivationHash, redeemScript);
+
+        // assert
+        List<ScriptChunk> originalRedeemScriptChunks = getOriginalRedeemScriptChunks(redeemScriptWithFlyoverDerivationHash);
+        assertEquals(redeemScript.getChunks(), originalRedeemScriptChunks);
+
+        List<ScriptChunk> flyoverChunks = getFlyoverChunks(redeemScriptWithFlyoverDerivationHash);
+        ScriptChunk flyoverDerivationHashChunk = flyoverChunks.get(0);
+        ScriptChunk opDropChunk = flyoverChunks.get(1);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImplTest.java:64]: You could get the first and second chunk directly here from `redeemScriptWithFlyoverDerivationHash`. No need to call `getFlyoverChunks`

Just an idea

File: rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilder.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package co.rsk.peg.bitcoin;
+
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.crypto.Keccak256;
+
+public interface FlyoverRedeemScriptBuilder {
+    Script addFlyoverDerivationHashToRedeemScript(Keccak256 flyoverDerivationHash, Script redeemScript);
```
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilder.java:7]: i thought about that, but the other builders receive keys and stuff to create the redeem script. I think this name is more accurate for what the method is actually doing. wdyt? I can change it anyways, no big deal

File: rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImpl.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            .data(flyoverDerivationHashSerialized)
+            .op(OP_DROP_CODE)
+            .addChunks(redeemScript.getChunks())
+            .build();
+    }
+
+    private void validateFlyoverDerivationHash(Keccak256 flyoverDerivationHash) {
+        if (isNull(flyoverDerivationHash) || flyoverDerivationHash.equals(zeroHash)) {
+            String message = "Provided flyover derivation hash is invalid.";
+            logger.warn("[validateFlyoverDerivationHash] {} {}", message, flyoverDerivationHash);
```
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImpl.java:32]: thanks!

File: rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImpl.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static co.rsk.peg.bitcoin.FlyoverRedeemScriptCreationException.Reason.INVALID_FLYOVER_DERIVATION_HASH;
+import static java.util.Objects.isNull;
+
+public class FlyoverRedeemScriptBuilderImpl implements FlyoverRedeemScriptBuilder {
+    private static final Logger logger = LoggerFactory.getLogger(FlyoverRedeemScriptBuilderImpl.class);
+    private static final Keccak256 zeroHash = Keccak256.ZERO_HASH;
+    private static final int OP_DROP_CODE = 117;
```
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImpl.java:15]: thanks!

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImplTest.java (hunk 3 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImplTest.java:64]: you right, thanks

File: rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptCreationException.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptCreationException.java]: done

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**